### PR TITLE
WIP: Fuseswap backend v1

### PR DESCRIPTION
**Pull request by mul53**

### CODE REVIEW

File: config/default.js [hunk 1]
```diff
@@ -13,5 +13,6 @@ module.exports = {
     sendgrid: {
       templates: {}
     }
-  }
+  },
+  rpc_url: 'https://rpc.fuse.io'
```
Comment (leonprou) [inline config/default.js:17]: please do not use cammelCase :)

File: src/controllers/api/v1/swap/index.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import { Request, Response, NextFunction } from 'express'
+import { Container } from 'typedi'
+import { validationResult } from 'express-validator'
+import SwapService from '../../../../services/swap'
```
Comment (leonprou) [inline src/controllers/api/v1/swap/index.ts:5]: you can use `module-alias`  to define root or src dirs.
so you can do like `@/services/swap`

**Review by leonprou — COMMENTED** [review]
Looks good and well structured 👍 
Gave you some comments

File: config/default.js (hunk 1 above)
Comment (leonprou) [reply config/default.js:17]: @mul53 sorry, meant use only `cammelCase`

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+name: Tests
```
Comment (leonprou) [inline .github/workflows/tests.yml:1]: cool 👍

File: .eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{
+  "parser": "@typescript-eslint/parser",
```
Comment (leonprou) [inline .eslintrc.json:2]: we use usually https://standardjs.com/ for our projects.
is it possible to use this with TS? any pros and cons of this config?

**Comment by mul53** [comment]
@leonprou made the changes on the new PR

**Pull request closed without merging**